### ISSUE TITLE
handshake capture does not contain valid handshake .

### ISSUE THREAD

**Issue by minanagehsalalma** [wifiphisher/wifiphisher#735]
make aircrack-ng verfiy the handshake  because it shows `handshake capture does not contain valid handshake` 

the fix in fluxion be like 
![image](https://user-images.githubusercontent.com/20546638/30241428-74be0fca-9583-11e7-9067-735005c79007.png)

**Comment by anakin1028** [comment]
I'm not sure I have captured what you mean. Do you mean our tool says that the capture you take doesn't capture fourway-handshake but other tools find that?

From the picture you post fluxion also says `Corrupted handshake`?

Can you post the `pcap` file you have captured in this thread?

**Comment by usama7628674** [comment]
He's saying that add a feature that checks for handshake whether it's valid or not like in Fluxion.

**Comment by usama7628674** [comment]
Fluxion says `corrupted handshake` when Handshake becomes corrupted.

**Comment by anakin1028** [comment]
When provide the `-hC` option for the `pcap` if victims enter the wrong pparaphrase we'll let the victims enter the paraphrase again.

The command would like:
`wifiphisher -hC xx.pcap` where xx.pcap stores the fourway handshake for the target access point.

**Comment by anakin1028** [comment]
So this means the handshake is not valid right?

**Comment by usama7628674** [comment]
Yes,absolutely.

**Comment by anakin1028** [comment]
That's what I don't get it... Our tool behaves the same as fluxion. From the description of the post we also output this handshake as invalid handshake?

**Comment by usama7628674** [comment]
I think he's using old fluxion

**Comment by usama7628674** [comment]
Maybe he's using fluxion handshake file with wifiphisher cuz sometimes handshake capture from fluxion
becomes corrupted.

**Comment by anakin1028** [comment]
@usama7628674 thanks : )

I'm not aware the behavior of the fluxion but the following is functionality of wifiphisher for `-hC` option:
`-hC xx.pcap` the `xx.pcap` must be prepared before running wifiphisher and after running wifiphisher wifiphisher will do the sanity check for this pcap file.

If the pcap file doesn't contain any fourway handshake the program will just exit and output that the handshake doesn't contain the valid handshake. Currently wifiphisher will not capture the handshake on the fly and the handshake file must be prepared before running the program.

After the victims has successfully connected to our fakeAP and enter the passphrase of the target access point, then wifiphisher will use the `pre-captured file` and the `passphrase` from the victim as input to validate if this passphrase is valid. If the passphrase is not valid we'll request victim to enter passphrase again, and if it's valid the behavior is just like before.

**Comment by usama7628674** [comment]
I think he didn't capture handshake and hoping that wifiphisher will capture it for him.That's what is happening to him.Wifiphisher requires handshake file before running it.

**Comment by minanagehsalalma** [comment]
@anakin1028  @usama7628674 no. i meant although the cap file content a valid handshake it says it's corrupted. and it's the same in fluxion but the fix it fluxion is to sanity check the handshake with aircrack-ng
![untitled](https://user-images.githubusercontent.com/20546638/30421837-5346fb04-993e-11e7-98f1-2fbc6a95a2b5.png)

**Comment by anakin1028** [comment]
Can you post the pcap files?

**Comment by minanagehsalalma** [comment]
@anakin1028 here it is http://rgho.st/67zZFn2Bp

**Comment by anakin1028** [comment]
is the link available?

**Comment by minanagehsalalma** [comment]
Yes

**Comment by anakin1028** [comment]
I can access the link you provide but I cannot download the file you posted inside that link.

**Comment by minanagehsalalma** [comment]
@anakin1028  why not .. here is a new link https://ufile.io/49o40 .

**Comment by anakin1028** [comment]
@minanagehsalalma thanks, the new link is avalilable for me.

I have seen your capture since the capture doesn't contain all the four messages in the fourway handshake messages so `wifiphisher` will judge it as the invalid handshake. The following figure is the screenshot from your pcap file. There are only `4-1` and `4-2` in the capture but without `4-3` and `4-4`.

![image](https://user-images.githubusercontent.com/6182530/30481862-476a438e-9a53-11e7-9ebd-e644e13e06b0.png)

Can you try to capture a handshake with all four messages are sniffed and try running wifiphisher again?

Maybe we only need `4-1` and `4-2` to complete this feature but I think it's not a serious constraint for collecting all four messages in `wifiphisher` (Our program will use `4-4` to calculate the MIC and uses the nonce values in `4-1` and `4-2`).

**Comment by minanagehsalalma** [comment]
But aircrack does with the 4-4 4-3 only and all of my handshakes didn't
work

**Comment by anakin1028** [comment]
I'm not aware about the implementation of aircrack but for `wifiphisher` you may try to capture all  the four frames and see if the tool works for you.

**Comment by minanagehsalalma** [comment]
but the most of the time it's hard to capture all the four frames ... because of the client is far away and the sometimes i use handshaker and familiar tools  to capture  a handshake

**Comment by anakin1028** [comment]
Have you used `wireshark` or `tcpdump` to capture? 

I may label this issue as an enhancement as follows

  * Release the constraint collecting all the four messages but only needs two (i.e. from `STA->AP` and `AP-> STA`). The possibility may becomes `4-1 and 4-2`, `4-1 and 4-4`, `4-3 and 4-4` and `4-2 and 4-3`
  * calculate MIC for `4-2`, or `4-3`, or `4-4` depends on which scenario you have collected in your tool.

But currently if you want to proceed with `wiifphisher` you need to collect all the four messages.

**Comment by minanagehsalalma** [comment]
@anakin1028  so you  will fix this ?  i Have used  airodump-ng to capture it . it's the most common way

**Comment by anakin1028** [comment]
Yea. This will be improved in the future.

**Comment by minanagehsalalma** [comment]
Can you specify a date. And Why not  now it's really important.

**Comment by sophron** [comment]
@minanagehsalalma Wifiphisher developers are volunteers. Asking "why not now?" is the same as demanding a present from a stranger on the street.

**Comment by minanagehsalalma** [comment]
i  am sorry 😢😢😢 i didn't mean this . I just wish that your tool become the best in it's job ...

**Comment by sophron** [comment]
@minanagehsalalma No worries. And thanks for contributing. :)

**Comment by minanagehsalalma** [comment]
@anakin1028 @sophron @usama7628674 
Now with pmkid around  ... can we expect a fix ?

**Comment by usama7628674** [comment]
@minanagehsalalma Yes cuz pkmid does not require clients to be connected to capture handshakes.

**Comment by minanagehsalalma** [comment]
@usama7628674  so does it support incomplete handshakes and pmkids now ?

but the tool Depends on clients .
pmkids wouldn't help much here
